### PR TITLE
Implement icon tint color effect

### LINKED ISSUE #496: [Enhancement] Implement IconTintColorEffect for Image and ImageButton
## Summary

To be able to better support theming, or even for whitelabeling your apps, I would like to have an IconTintColorEffect. With this effect you could tint an Image or ImageButton without needing to add your icon in multiple colors to your project.

## API Changes

Added `IconTintColor` Effect which holds an attached property `Xamarin.Forms.Color TintColor`.
When this attached property is set, the effect will automatically get added to the view.

Implemented a shared `IconTintColorEffectRouter` and implemented an Android and iOS specific `IconTintColorEffectRouter` to implement this effect for these platforms.

## Intended Use Case
If your app supports theming, you already easily can change colors on almost anything, except for your icons that you might use in an Image control or ImageButton. With this effect you can now easily change the color of your icons as well.

## Who Will Do The Work?

- [x] I am willing to take this on myself
- [ ] Just putting this out there for someone to pick up

### PR DESCRIPTION
**Pull request by jBijsterboschNL**
### Description of Change ###

Implemented IconTintColorEffect for iOS and Android to set a tint color to Image and ImageButton controls.

### Bugs Fixed ###

- Fixes #496

### API Changes ###

Added TintColor attached property for Image and ImageButton controls. When setting this attached property, the IconTintColorEffect is added to the control. The effect is implemented for the iOS and Android platforms.

### Behavioral Changes ###

N/A

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard
- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Comment by jBijsterboschNL** [comment]
I'm still working on the iOS implementation. Hope to get this pushed today, but otherwise somewhere next week.

**Comment by jBijsterboschNL** [comment]
Darn incoming commits... 😛

File: XamarinCommunityToolkit/Effects/IconTintColor/IconTintColorEffectRouter.android.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+				case ImageView image:
+					image.ClearColorFilter();
+					break;
+				case Button button:
+					var drawables = button.GetCompoundDrawables().Where(d => d != null);
+
+					foreach (var img in drawables)
+					{
+						img.ClearColorFilter();
+					}
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Effects/IconTintColor/IconTintColorEffectRouter.android.cs:64]:
```suggestion
						img.ClearColorFilter();
```

File: XamarinCommunityToolkit/Effects/IconTintColor/IconTintColorEffectRouter.android.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+					break;
+			}
+		}
+
+		void SetImageViewTintColor(ImageView image, Forms.Color color)
+		{
+			if (color == Forms.Color.Default)
+			{
+				image.ClearColorFilter();
+			}
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Effects/IconTintColor/IconTintColorEffectRouter.android.cs:74]:
```suggestion
				image.ClearColorFilter();
```

File: XamarinCommunityToolkit/Effects/IconTintColor/IconTintColorEffectRouter.android.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+		void SetButtonTintColor(Button button, Forms.Color color)
+		{
+			var drawables = button.GetCompoundDrawables().Where(d => d != null);
+
+			if (color == Forms.Color.Default)
+			{
+				foreach (var img in drawables)
+				{
+					img.ClearColorFilter();
+				}
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Effects/IconTintColor/IconTintColorEffectRouter.android.cs:88]:
```suggestion
					img.ClearColorFilter();
```

File: XamarinCommunityToolkit/Effects/IconTintColor/IconTintColorEffectRouter.android.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+			}
+
+			foreach (var img in drawables)
+			{
+				img.SetTint(color.ToAndroid());
+			}
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Effects/IconTintColor/IconTintColorEffectRouter.android.cs:94]:
```suggestion
				img.SetTint(color.ToAndroid());
```

**Review by AndreiMisiukevich — COMMENTED** [review]
Could you fix the code format a little?)
Typically we don't use parentheses in one-line operator's bodies

**Comment by jBijsterboschNL** [comment]
> Could you fix the code format a little?)
> Typically we don't use parentheses in one-line operator's bodies

Hi @AndreiMisiukevich, of course no problem, my bad. I'm used to use braces even white single line statements, same for not using a lambda for single line methods. So sometimes I forget, which isn't consistent 😉  thanks for checking this! I've re-checked my entire code change, could you review again? 😃

**Review by jfversluis — APPROVED** [review]

**Pull request merged**